### PR TITLE
[MNG-7385] improve repository metadata documentation

### PR DESCRIPTION
**Pull request by hboutemy**
https://issues.apache.org/jira/browse/MNG-7385

### CODE REVIEW

File: maven-repository-metadata/src/main/mdo/metadata.mdo [hunk 1]
```diff
@@ -338,10 +339,14 @@ under the License.
...
       </fields>
     </class>
+
     <class java.clone="deep">
       <name>Plugin</name>
       <version>1.0.0+</version>
-      <description>Mapping information for a single plugin within this group</description>
+      <description>Mapping information for a single plugin within this group (deprecated).</description>
+      <annotations>
+        <annotation>@Deprecated</annotation>
```
Comment (michael-o) [inline maven-repository-metadata/src/main/mdo/metadata.mdo:348]: Who will actually perform the import for this annotation?

Comment (hboutemy) [reply maven-repository-metadata/src/main/mdo/metadata.mdo:348]: "perform the import for this annotation"? what does that mean?
with this annotation, the class is marked deprecated (in generated java): I suppose nobody uses it apart the plugin-tools and Sonatype Nexus Staging
but what is useful is that it documents deprecation of data, given update of the data will disappear in the future with maven-compat removal

Comment (michael-o) [reply maven-repository-metadata/src/main/mdo/metadata.mdo:348]: I mean the Java import for the annotation class. Where is it? Will Modello itself generate the import statement?

Comment (hboutemy) [reply maven-repository-metadata/src/main/mdo/metadata.mdo:348]: @Deprecated does not require import: I tested, generated java compiles, no problem

**Review by michael-o — COMMENTED** [review]
Why did you change the order in the Modello model?

**Comment by hboutemy** [comment]
for sake of clarity in the generated documentation: I grouped fields to show which fields were available in which context (groupid, groupid/artifactid, groupid/artifactid/version)

**Comment by hboutemy** [comment]
last call for improvement before merging to master and maven-3.8.x branches

File: maven-repository-metadata/src/site/apt/index.apt [hunk 2]
```diff
@@ -27,26 +27,32 @@ Maven Repository Metadata Model
...
+ the Maven Repository Metadata file contains 3 different sets of metadata:
 
- * <<<maven-metadata.xml>>> in a remote repository,
+ [[1]] in a "groupId" directory: a "groupId" directory may contain Maven plugins artifacts, which are described in metadata's <<<plugins>>> element,
 
- * <<<maven-metadata-\<repo-id>.xml>>> in a local repository, for metadata from a repository with <<<repo-id>>> identifier.
+ [[2]] in a "groupId/artifactId" directory: metadata describes <<<groupId>>>, <<<artifactId>>> and <<<versioning>>> element that
+       gives data about available versions (<<<latest>>>, <<<release>>>, <<<versions>>> list and <<<lastUpdated>>>),
+
+ [[3]] in a "groupId/artifactId/version" snapshot artifact directory: metadata describes <<<groupId>>>, <<<artifactId>>>, <<<version>>> (base version, ie ending in <<<-SNAPSHOT>>>) and
```
Comment (michael-o) [inline maven-repository-metadata/src/site/apt/index.apt:46]: ie => i..e.

File: maven-repository-metadata/src/site/apt/index.apt [hunk 3]
```diff
@@ -27,26 +27,32 @@ Maven Repository Metadata Model
...
 
  The following are generated from this model:
 
-   * {{{./apidocs/index.html}Java sources}} with Reader and Writers for the Xpp3 XML parser, to read and write <<<maven-metadata(-*).xml>>> files
+   * {{{./apidocs/index.html}Java sources}} with Reader and Writers for the Xpp3 XML parser, to read and write <<<maven-metadata(-*).xml>>> files,
 
-   * A {{{./repository-metadata.html}Descriptor Reference}}
+   * a {{{./repository-metadata.html}Descriptor Reference}}.
+ 
+ Notice: data about plugins in a directory representing a groupId is deprecated and will be removed in a future Maven version (perhaps 4)
```
Comment (michael-o) [inline maven-repository-metadata/src/site/apt/index.apt:58]: Is there a JIRA issue for this?

Comment (hboutemy) [reply maven-repository-metadata/src/site/apt/index.apt:58]: it's a consequence of removing maven-compat: I don't know the MNG for it

Comment (michael-o) [reply maven-repository-metadata/src/site/apt/index.apt:58]: Maybe you should just mention that.

Comment (hboutemy) [reply maven-repository-metadata/src/site/apt/index.apt:58]: or more precisely: removing maven-compat will stop updating the value
removing the field is IMHO more clear than having half updated values

Comment (hboutemy) [reply maven-repository-metadata/src/site/apt/index.apt:58]: I can remove ht "maybe 4" part that is both too vague and too precise

Comment (gnodet) [reply maven-repository-metadata/src/site/apt/index.apt]: > it's a consequence of removing maven-compat: I don't know the MNG for it

https://issues.apache.org/jira/browse/MNG-7266 ?

**Comment by hboutemy** [comment]
merged in https://github.com/apache/maven/commit/1af8513fa7512cf25022b249cae0f84062c5085b

**Comment by jira-importer** [comment]
Resolve #9178

**Pull request closed without merging**